### ISSUE TITLE
Amplify publish ignores the project-config.json's BuildCommand

### ISSUE THREAD

**Issue by manu2504** [aws-amplify/amplify-cli#8059]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

12.22.1

### Amplify CLI Version

5.3.1

### What operating system are you using?

Mac

### Amplify Categories

hosting

### Amplify Commands

publish

### Describe the bug

`amplify publish` tries to build the project with command `ng run app:build` while the build command specified in .config/project-config.json file is `"BuildCommand": "ionic build"`. Complete file:
```json
{
  "providers": [
    "awscloudformation"
  ],
  "projectName": "sodexoApp",
  "version": "3.1",
  "frontend": "javascript",
  "javascript": {
    "framework": "ionic",
    "config": {
      "SourceDir": "src",
      "DistributionDir": "www",
      "BuildCommand": "ionic build",
      "StartCommand": "ionic serve"
    }
  }
}
```

### Expected behavior

Amplify should build project with `ionic build` command.

### Reproduction steps

1. `mkdir amplify-test && cd amplify-test`
2. `npm init`
3. `ionic init`
4. `amplify init`
? Initialize the project with the above configuration? No
? Enter a name for the environment dev
? Choose your default editor: Visual Studio Code
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using ionic
? Source Directory Path:  src
? Distribution Directory Path: www
? Build Command:  ionic build
? Start Command: ionic serve
5. `amplify push`
6. `amplify publish`
Error will be thrown here.

You may also clone https://github.com/manu2504/amplify-test 

### Log output

<details>

```
# Put your logs below this line
amplify publish             
✔ Successfully pulled backend environment dbs from the cloud.

    Current Environment: dbs
    
┌──────────┬─────────────────┬───────────┬───────────────────┐
│ Category │ Resource name   │ Operation │ Provider plugin   │
├──────────┼─────────────────┼───────────┼───────────────────┤
│ Auth     │ sodexoappauth   │ No Change │ awscloudformation │
├──────────┼─────────────────┼───────────┼───────────────────┤
│ Hosting  │ S3AndCloudFront │ No Change │ awscloudformation │
└──────────┴─────────────────┴───────────┴───────────────────┘

No changes detected
> ng run app:build

[INFO] Looks like @angular/cli isn't installed in this project.
       
       This package is required for this command to work properly.

? Install @angular/cli? [Y/n]

```

</details>

**Comment by cjihrig** [comment]
I think this is related to your use of ionic. From what I can tell, the Amplify CLI is running `ionic build`, and the error you're seeing is coming from there.

If I run through this issue without installing ionic, I see an error while running `ionic build`:

```
Error: spawn ionic ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
    at onErrorNT (internal/child_process.js:467:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn ionic',
  path: 'ionic',
  spawnargs: [ 'build' ]
}
```

Additionally, the error message you're seeing can be reproduced without the Amplify CLI:

```
$ npx ionic build
> ng run app:build

[INFO] Looks like @angular/cli isn't installed in this project.
       
       This package is required for this command to work properly.

? Install @angular/cli? No
[WARN] Not installing--here's how to install manually: npm i -D -E @angular/cli


  ────────────────────────────────────────────────────────────

          Ionic CLI update available: 5.4.16 → 6.17.0
                                
     The package name has changed from ionic to @ionic/cli!
                                
             To update, run: npm uninstall -g ionic
                 Then run: npm i -g @ionic/cli

  ────────────────────────────────────────────────────────────

[ERROR] @angular/cli is required for this command to work properly.
```

**Comment by manu2504** [comment]
Indeed, thanks for the extra check!
